src/pages/AllDao.vue

```diff
@@ -233,54 +301,104 @@ export default defineComponent({
 			setSelectedKeys(e.target.value ? [e.target.value] : []);
 		},
 		handleFilterData(value: string, record: DaoTableData) {
+			// TODO: update to use filter from backend (for name)
 			return record.name.toString().toLowerCase().includes(value.toLowerCase());
 		},
-	},
-	setup() {
-		const formState = reactive(DaoStore());
-		const state = reactive({
-			searchText: "",
-			searchedColumn: "",
-		});
-		const searchInput = ref();
-		const tempArray: Array<DaoTableData> = [];
-		formState.psqlData = tempArray; // prohibit duplication of data
-		// Get all daos
-		// Here 40 --> Page size
-		executeReq(getAllDaoReq(40, null, null, null))
-			.then((res) => {
-				if (res && res.allSigmaDaos && res.allSigmaDaos.nodes.length) {
+		async getCursorDetails(pageNumber: number) {
+			const cursorRes = await executeReq(
+				getCursorReq(pageNumber, ROWS_PER_PAGE)
+			).catch((error) =>
+				openErrorNotificationWithIcon(UNSUCCESSFUL, error.message)
+			);
+
+			if (
+				cursorRes &&
+				cursorRes.allSigmaDaos &&
+				cursorRes.allSigmaDaos.pageInfo
+			) {
+				const pageInfo = cursorRes.allSigmaDaos.pageInfo;
+				this.currentPageCursor.endCursor = pageInfo.endCursor;
+				this.currentPageCursor.startCursor = pageInfo.startCursor;
+			}
+		},
+		async fetchDataForDAO(
+			first: number | null,
+			endCursor: string | null,
+			last: number | null,
+			startCursor: string | null,
+			currentPage?: number
+		) {
+			const res = await executeReq(
+				getAllDaoReq(first, endCursor, last, startCursor)
+			).catch((error) =>
+				openErrorNotificationWithIcon(UNSUCCESSFUL, error.message)
+			);
+			if (res && res.allSigmaDaos) {
+				if (res.allSigmaDaos.nodes.length) {
+					// clean existing data in temp array with change of page
+					if (this.dataSource.length) {
+						this.dataSource = [];
+					}
 					res.allSigmaDaos.nodes.map(async (item: any, index: number) => {
 						if (item.appParams) {
 							item.appParams = JSON.parse(item.appParams);
 						}
 						const globalState = decodeAppParamsState(item.appParams.dt.gd);
 						const tokenData = await getAssetInformation(item.assetId);
-						formState.psqlData.push({
+						let data = {
 							key: index,
-							dao_id: item.appId,
-							token_id: item.assetId,
+							dao_id: +item.appId,
+							token_id: +item.assetId,
 							token_name: tokenData.name as string,
 							name: globalState.get(GLOBAL_STATE_MAP_KEY.DaoName) as string,
 							link: globalState.get(GLOBAL_STATE_MAP_KEY.Url) as string,
-						});
+						};
+						this.dataSource.push(data);
+
+						// pushing data to store only if it doesn't exists
+						let dataExists: boolean;
```

Comment:
I would suggest this.
```suggestion
						let isCached: boolean;
```